packages/compat/src/keypair.ts

```diff
@@ -0,0 +1,63 @@
+import { Keypair } from '@solana/web3.js';
+
+function addPkcs8Header(bytes: Uint8Array): Uint8Array {
+    // prettier-ignore
+    const header = new Uint8Array([
+        /**
+         * PKCS#8 header
+         */
+        0x30, // ASN.1 sequence tag
+        0x2e, // Length of sequence (46 more bytes)
+
+            0x02, // ASN.1 integer tag
+            0x01, // Length of integer
+                0x00, // Version number
+
+            0x30, // ASN.1 sequence tag
+            0x05, // Length of sequence
+                0x06, // ASN.1 object identifier tag
+                0x03, // Length of object identifier
+                    // Edwards curve algorithms identifier https://oid-rep.orange-labs.fr/get/1.3.101.112
+                        0x2b, // iso(1) / identified-organization(3) (The first node is multiplied by the decimal 40 and the result is added to the value of the second node)
+                        0x65, // thawte(101)
+                    // Ed25519 identifier
+                        0x70, // id-Ed25519(112)
+    ]);
+    const result = new Uint8Array(header.length + bytes.length);
+    result.set(header);
+    result.set(bytes, header.length);
+    return result;
+}
+
+/**
+ * Convert from a Legacy Web3 JS Keypair to a CryptoKeyPair
+ * @param keypair   The Keypair to convert
+ * @returns         A CryptoKeyPair
+ */
+export async function fromLegacyKeypair(keypair: Keypair): Promise<CryptoKeyPair> {
+    const publicKey: CryptoKey = await crypto.subtle.importKey(
+        'raw',
+        keypair.publicKey.toBytes(),
+        {
+            name: 'EdDSA',
+            namedCurve: 'Ed25519',
+        },
+        true,
+        ['verify']
+    );
+    const privateKeyBytesPkcs8 = addPkcs8Header(keypair.secretKey);
+    const privateKey: CryptoKey = await crypto.subtle.importKey(
+        'pkcs8',
+        privateKeyBytesPkcs8,
+        {
+            name: 'EdDSA',
+            namedCurve: 'Ed25519',
+        },
+        false,
```

Comment:
With `generateKey` we allow the private key to have `extractable` set by the caller. I'm not 100% sure this is correct (https://github.com/solana-labs/solana-web3.js/issues/1723), but if it is then I wonder if we should have the same here?

Reply:
I was trying to dig for an answer to this issue, because I swear I read somewhere that it will reject you if you try to import a private key and mark it extractable, but I can't be sure. Maybe we just need to try it.

Reply:
added the latter, for review